x-pack/plugins/cases/public/components/actions/assignees/edit_assignees_selectable.test.tsx

```diff
@@ -0,0 +1,413 @@
+
+    await waitFor(() => {
+      expect(result.getByText('Damaged Raccoon'));
+    });
+
+    expect(result.queryByTestId(searchedUserDataTestSubj)).not.toBeInTheDocument();
+
+    await waitForComponentToUpdate();
+  });
+
+  it('does not show the same user in search results if it is already in the initial user profile mapping', async () => {
+    const result = appMock.render(<EditAssigneesSelectable {...props} />);
+
+    userEvent.type(result.getByPlaceholderText('Search'), 's');
+
+    act(() => {
+      jest.advanceTimersByTime(1000);
+    });
+
+    const searchResults = result.getAllByTestId(
+      // Physical Dinosaur
+      'cases-actions-assignees-edit-selectable-assignee-u_A_tM4n0wPkdiQ9smmd8o0Hr_h61XQfu8aRPh9GMoRoc_0-icon-empty'
+    );
+
+    expect(searchResults.length).toBe(1);
+
+    await waitForComponentToUpdate();
+  });
+
+  it('selects a searched user correctly', async () => {
+    // Silly Hare
+    const searchedUserDataTestSubj =
+      'cases-actions-assignees-edit-selectable-assignee-u_IbBVXpDtrjOByJ-syBdr425fLGqwpzY_xdQqCFAFXLI_0';
+
+    const result = appMock.render(<EditAssigneesSelectable {...props} />);
+
+    userEvent.type(result.getByPlaceholderText('Search'), 's');
+
+    act(() => {
+      jest.advanceTimersByTime(1000);
+    });
+
+    await waitFor(() => {
+      expect(result.getByTestId(searchedUserDataTestSubj));
+    });
+
+    userEvent.click(result.getByTestId(searchedUserDataTestSubj));
+    expect(props.onChangeAssignees).toBeCalledWith({
+      selectedItems: [
+        'u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0',
+        'u_IbBVXpDtrjOByJ-syBdr425fLGqwpzY_xdQqCFAFXLI_0',
+      ],
+      unSelectedItems: [],
+    });
+
+    await waitForComponentToUpdate();
+  });
+
+  it('shows deselected users from the initial user profile mapping', async () => {
+    const result = appMock.render(<EditAssigneesSelectable {...props} />);
+
+    // @ts-ignore: full name exists
+    userEvent.click(result.getByText(userProfiles[0].user.full_name));
+
+    // @ts-ignore: full name exists
+    expect(result.getByText(userProfiles[0].user.full_name)).toBeInTheDocument();
+    // ensures that the icon is set to empty
+    expect(
+      result.getByTestId(
+        'cases-actions-assignees-edit-selectable-assignee-u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0-icon-empty'
+      )
+    ).toBeInTheDocument();
+
+    expect(props.onChangeAssignees).toBeCalledWith({
+      selectedItems: [],
+      unSelectedItems: ['u_J41Oh6L9ki-Vo2tOogS8WRTENzhHurGtRc87NgEAlkc_0'],
+    });
+
+    await waitForComponentToUpdate();
+  });
+
+  it('does not shows initial empty search results on the list of users', async () => {
+    // Silly Hare
+    const searchedUserDataTestSubj =
+      'cases-actions-assignees-edit-selectable-assignee-u_IbBVXpDtrjOByJ-syBdr425fLGqwpzY_xdQqCFAFXLI_0';
+
+    const result = appMock.render(<EditAssigneesSelectable {...props} />);
+
+    act(() => {
+      jest.advanceTimersByTime(1000);
```

Comment:
The suggestion hook uses a debounce hook to prevent fetching as the user types. We need to advance the timer so the `onDebounce` hook triggers. We could wait but it is better if we have control over the timers, denounces, etc as we can avoid timeouts.

Reply:
But doesn't the debounce only apply to the search/typing?

In this case, we are looking for a case with an already selected user so it should already be there anyway. (no debounce applied?)

Reply:
The suggestion API will run on the first render with an empty string as a search term (it is designed by purpose as it). I want to be sure that after the end of debouncing where the results are returned from the suggestion API. Is there any way I can test that without forwarding the time?

Reply:
Ah, wasn't aware of this

> The suggestion API will run on the first render with an empty string as a search term

All good then 🙌